crates/revive-strategy/src/cheatcodes/mod.rs

```diff
@@ -164,6 +167,11 @@ fn set_timestamp(new_timestamp: U256, ecx: Ecx<'_, '_, '_>) {
     });
 }
 
+fn set_basefee(new_basefee: U256, ecx: Ecx<'_, '_, '_>) {
+    // Set basefee in EVM context.
+    ecx.block.basefee = new_basefee.try_into().expect("Basefee exceeds u64");
```

Comment:
did you check how zksync is handling it?
I think they keep it consistent between vm calls.

  1. Single Source of Truth

  // vm.fee() cheatcode sets this:
  ccx.ecx.block.basefee =
  newBasefee.saturating_to();  // Line 459 in 
  evm.rs

  2. Shared Through EthEvmContext

  When zkVM is called, it receives the same
  EthEvmContext that EVM uses:

  // In zksync_try_call (strategy/zksync):
  foundry_zksync_core::vm::call(call,
  factory_deps, ecx, ccx)


  3. zkVM Extracts Basefee from Shared Context

  // In runner.rs:303 - zkVM reads basefee from 
  the shared ecx.block:
  block_basefee: min(max_fee_per_gas.to_ru256(),
  rU256::from(ecx.block.basefee)),